lib/Db/ItemMapperV2.php

```diff
@@ -164,4 +195,302 @@ public function purgeDeleted(?string $userID, ?int $oldestDelete): void
+            ->andWhere('feeds.user_id = :userId')
+            ->andWhere('folders.id = :folderId')
+            ->setParameters(['updatedSince' => $updatedSince, 'folderId' => $folderId, 'userId' => $userId])
+            ->orderBy('items.updated_date', 'DESC');
+
+        if ($hideRead === true) {
+            $builder->andWhere('items.unread = 1');
+        }
+
+        return $this->findEntities($builder);
+    }
+
+    /**
+     * @param string $userId
+     * @param int    $updatedSince
+     * @param int    $feedType
+     *
+     * @return Item[]|Entity[]
+     * @throws ServiceValidationException
+     */
+    public function findAllAfter(string $userId, int $feedType, int $updatedSince): array
+    {
+        $builder = $this->db->getQueryBuilder();
+
+        $builder->select('items.*')
+            ->from($this->tableName, 'items')
+            ->innerJoin('items', FeedMapperV2::TABLE_NAME, 'feeds', 'items.feed_id = feeds.id')
+            ->andWhere('items.updated_date >= :updatedSince')
+            ->andWhere('feeds.user_id = :userId')
+            ->setParameters(['updatedSince' => $updatedSince, 'userId' => $userId])
+            ->orderBy('items.updated_date', 'DESC');
+
+        switch ($feedType) {
+            case FeedType::STARRED:
+                $builder->andWhere('items.starred = 1');
+                break;
+            case FeedType::UNREAD:
+                $builder->andWhere('items.unread = 1');
+                break;
+            default:
+                throw new ServiceValidationException('Unexpected Feed type in call');
+        }
+
+        return $this->findEntities($builder);
+    }
+
+    /**
+     * @param string $userId
+     * @param int    $feedId
+     * @param int    $limit
+     * @param int    $offset
+     * @param bool   $hideRead
+     * @param bool   $oldestFirst
+     * @param array  $search
+     *
+     * @return Item[]
+     */
+    public function findAllFeed(
+        string $userId,
+        int $feedId,
+        int $limit,
+        int $offset,
+        bool $hideRead,
+        bool $oldestFirst,
+        array $search
+    ): array {
+        $builder = $this->db->getQueryBuilder();
+
+        $builder->select('items.*')
+            ->from($this->tableName, 'items')
+            ->innerJoin('items', FeedMapperV2::TABLE_NAME, 'feeds', 'items.feed_id = feeds.id')
+            ->andWhere('feeds.user_id = :userId')
+            ->andWhere('items.feed_id = :feedId')
+            ->setParameter('userId', $userId)
+            ->setParameter('feedId', $feedId)
+            ->setMaxResults($limit)
+            ->setFirstResult($offset)
+            ->orderBy('items.updated_date', ($oldestFirst ? 'ASC' : 'DESC'));
+
+        if ($search !== []) {
+            foreach ($search as $key => $term) {
+                $term = $this->db->escapeLikeParameter($term);
+                $builder->andWhere("items.search_index LIKE :term${key}")
```

Comment:
we could use the `$builder->createNamedParameter($term)` here also, even though it's not that critical since `key` is a number. (same for L433 and L478)